Source/Core/Core/IOS/DI/DI.cpp

```diff
@@ -30,93 +52,658 @@ void DI::DoState(PointerWrap& p)
+    }
 
     const u64 partition_offset =
         static_cast<u64>(Memory::Read_U32(request.in_vectors[0].address + 4)) << 2;
-    const DiscIO::Partition partition(partition_offset);
-    DVDInterface::ChangePartition(partition);
+    ChangePartition(DiscIO::Partition(partition_offset));
 
-    INFO_LOG(IOS_DI, "DVDLowOpenPartition: partition_offset 0x%016" PRIx64, partition_offset);
+    INFO_LOG(IOS_DI, "DVDLowOpenPartition: partition_offset 0x%09" PRIx64, partition_offset);
 
     // Read TMD to the buffer
-    const IOS::ES::TMDReader tmd = DVDThread::GetTMD(partition);
+    const IOS::ES::TMDReader tmd = DVDThread::GetTMD(m_current_partition);
     const std::vector<u8>& raw_tmd = tmd.GetBytes();
     Memory::CopyToEmu(request.io_vectors[0].address, raw_tmd.data(), raw_tmd.size());
-    m_ios.GetES()->DIVerify(tmd, DVDThread::GetTicket(partition));
 
-    return_value = 1;
+    ReturnCode es_result = m_ios.GetES()->DIVerify(tmd, DVDThread::GetTicket(m_current_partition));
+    Memory::Write_U32(es_result, request.io_vectors[1].address);
+
+    return_value = DIResult::SUCCESS;
     break;
   }
+  case DIIoctl::DVDLowGetNoDiscOpenPartitionParams:
+    ERROR_LOG(IOS_DI, "DVDLowGetNoDiscOpenPartitionParams - dummied out");
+    DolphinAnalytics::Instance().ReportGameQuirk(GameQuirk::USES_DIFFERENT_PARTITION_COMMAND);
+    request.DumpUnknown(GetDeviceName(), LogTypes::IOS_DI);
+    break;
+  case DIIoctl::DVDLowNoDiscOpenPartition:
+    ERROR_LOG(IOS_DI, "DVDLowNoDiscOpenPartition - dummied out");
+    DolphinAnalytics::Instance().ReportGameQuirk(GameQuirk::USES_DIFFERENT_PARTITION_COMMAND);
+    request.DumpUnknown(GetDeviceName(), LogTypes::IOS_DI);
+    break;
+  case DIIoctl::DVDLowGetNoDiscBufferSizes:
+    ERROR_LOG(IOS_DI, "DVDLowGetNoDiscBufferSizes - dummied out");
+    DolphinAnalytics::Instance().ReportGameQuirk(GameQuirk::USES_DIFFERENT_PARTITION_COMMAND);
+    request.DumpUnknown(GetDeviceName(), LogTypes::IOS_DI);
+    break;
+  case DIIoctl::DVDLowOpenPartitionWithTmdAndTicket:
+    ERROR_LOG(IOS_DI, "DVDLowOpenPartitionWithTmdAndTicket - not implemented");
+    DolphinAnalytics::Instance().ReportGameQuirk(GameQuirk::USES_DIFFERENT_PARTITION_COMMAND);
+    break;
+  case DIIoctl::DVDLowOpenPartitionWithTmdAndTicketView:
+    ERROR_LOG(IOS_DI, "DVDLowOpenPartitionWithTmdAndTicketView - not implemented");
+    DolphinAnalytics::Instance().ReportGameQuirk(GameQuirk::USES_DIFFERENT_PARTITION_COMMAND);
+    break;
   default:
+    ERROR_LOG(IOS_DI, "Unknown ioctlv 0x%02x", request.request);
     request.DumpUnknown(GetDeviceName(), LogTypes::IOS_DI);
   }
-  return GetDefaultReply(return_value);
+  return GetDefaultReply(static_cast<s32>(return_value));
+}
+
+void DI::ChangePartition(const DiscIO::Partition partition)
+{
+  m_current_partition = partition;
+}
+
+DiscIO::Partition DI::GetCurrentPartition()
+{
+  auto di = IOS::HLE::GetIOS()->GetDeviceByName("/dev/di");
+  if (di)
```

Comment:
I don't think a guard clause makes sense here; either case is equally important (that code is hit on Gamecube mode when IOS doesn't exist, since the partition is used for `UpdateRunningGameMetadata`).

Reply:
Ah, that makes sense. Still, I think the convention is to not have an else after a return (IIRC... it's been a while since my last PR :P)

Reply:
I don't see anything about that in the contributing guide (the only example with return is [for multiline conditionals](https://github.com/dolphin-emu/dolphin/blob/master/Contributing.md#general), and the other examples for else [are also about lines](https://github.com/dolphin-emu/dolphin/blob/master/Contributing.md#conditionals)).  I haven't spotted any existing code that goes either way, though, so that might still just be the convention and I haven't seen it yet.

Reply:
Yeah, it's not a convention that's explicitly mentioned in the contributing guide. But I just remembered it from past PRs and a quick search led me to this comment by @lioncash: https://github.com/dolphin-emu/dolphin/pull/6034#discussion_r137761025

Older parts of the codebase routinely violate that convention, but that shouldn't be the case for new code.